geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/JdbcMetadata.scala

```diff
@@ -108,10 +108,19 @@ class JdbcMetadata(
   override def removePartition(partition: PartitionMetadata): Unit =
     WithClose(pool.getConnection())(connection => PartitionsTable.delete(connection, root, partition))
 
+  override def setPartitions(partitions: Seq[PartitionMetadata]): Unit = {
+    // TODO execute as a transaction
+    WithClose(pool.getConnection()) { connection =>
+      PartitionsTable.clear(connection, root)
+      partitions.foreach(PartitionsTable.insert(connection, root, _))
+    }
+  }
+
   // noinspection ScalaDeprecation
   override def compact(partition: Option[String], threads: Int): Unit = compact(partition, None, threads)
 
   override def compact(partition: Option[String], fileSize: Option[Long], threads: Int): Unit = {
+    // TODO execute as a transaction
```

Comment:
As usual, I'm on the fence about a ticket here.  To your point, if we aren't planning on it, no need.  Other the other hand, if we really ought to do it, it may be worth being in the backlog.

Reply:
yeah, we already had a bunch of TODOs in here to note places where a transaction should be added...

Reply:
+1.  Yeah, if all this is for transaction support, yeah, no need for a ticket.